### ISSUE TITLE
Timepicker: Hourformat not working correctly

### ISSUE THREAD

**Issue by harilalinapp** [primefaces/primeng#1446]
**I'm submitting a ...**  (check one with "x")
Problem with hourformat in calender

[ ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
<!-- Describe how the bug manifests. -->
When set to hourformat=12 the hour increment beyond 24. And also shows wrong time in input 
When opened the calender in 12 hour format, change the hour only not minutes.
**Expected behavior**
<!-- Describe what the behavior would be without the bug. -->
Set to hourformat12 it should show only 

**Minimal reproduction of the problem with instructions**
Also it would be better to set the stephour and stepminutes so that user can set the hour and minutes easily. 
<!--
If the current behavior is a bug or you can illustrate your feature request better with an example, 
please provide the *STEPS TO REPRODUCE* and if possible a *MINIMAL DEMO* of the problem via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).
-->

**What is the motivation / use case for changing the behavior?**
Changing the minutes one by one is nasty.
<!-- Describe the motivation or the concrete use case -->

**Please tell us about your environment:**
<!-- Operating system, IDE, package manager, HTTP server, ... -->
ubuntu 14

* **Angular version:** 2.0.X
<!-- Check whether this is still an issue in the most recent Angular version -->
angular-cli": "1.0.0-beta.21",
* **PrimeNG version:** 2.0.X
<!-- Check whether this is still an issue in the most recent Angular version -->

"primeng": "^1.0.0",
* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
<!-- All browsers where this could be reproduced -->
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by Mrtcndkn** [comment]
It's fixed for 1.0.1

**Comment by harilalinapp** [comment]
In editable datepicker it's causing problem. If the datepicker is set to 12hr format and manually editing the input field won't update the time.
